AppCenter/AppCenterTests/MSAuthTokenContextTests.m

```diff
@@ -418,4 +418,29 @@ - (void)testAuthTokenNeedsToBeRefreshed {
   OCMVerify([delegateMock authTokenContext:OCMOCK_ANY authTokenNeedsToBeRefreshed:expectedAccountId]);
 }
 
+- (void)testfinishDoesNotResetTokenIfInitialized {
+
+  // If
+  [self.sut doNotResetAuthTokenAfterStart];
+  OCMReject([self.sut setAuthToken:OCMOCK_ANY withAccountId:OCMOCK_ANY expiresOn:OCMOCK_ANY]);
+
+  // When
+  [self.sut finishInitialize];
+
+  // Then
+  OCMVerify(self.sut);
+}
+
+- (void)testfinishInsertsAnonymousSessionIfNotPrevented {
```

Comment:
For clarity and consistency with the previous test, maybe call these two tests:

> testFinishDoesNotResetTokenIfPrevented

> testFinishResetsTokenIfNotPrevented

PS: `Finish` not `finish`